apis/bigtable/v1beta1/bigtableinstance_types.go

```diff
@@ -0,0 +1,191 @@
+	// +optional
+	StorageTarget *int64 `json:"storageTarget,omitempty"`
+}
+
+type InstanceCluster struct {
+	/* A list of Autoscaling configurations. Only one element is used and allowed. */
+	// +optional
+	AutoscalingConfig *InstanceAutoscalingConfig `json:"autoscalingConfig,omitempty"`
+
+	/* The ID of the Cloud Bigtable cluster. Must be 6-30 characters and must only contain hyphens, lowercase letters and numbers. */
+	ClusterId string `json:"clusterId"`
+
+	// Describes the Cloud KMS encryption key that will be used to protect the destination Bigtable
+	// cluster. The requirements for this key are:
+	//
+	// 1) The Cloud Bigtable service account associated with the project that contains
+	// this cluster must be granted the cloudkms.cryptoKeyEncrypterDecrypter role on the CMEK key.
+	// 2) Only regional keys can be used and the region of the CMEK key must match the region of the cluster.
+	// 3) All clusters within an instance must use the same CMEK key access to this encryption key.
+	// +optional
+	KmsKeyRef *refs.KMSCryptoKeyRef `json:"kmsKeyRef,omitempty"`
+
+	/* The number of nodes in the cluster. If no value is set, Cloud Bigtable automatically allocates nodes based on your data footprint and optimized for 50% storage utilization. */
+	// +optional
+	NumNodes *int64 `json:"numNodes,omitempty"`
+
+	/* The storage type to use. One of "SSD" or "HDD". Defaults to "SSD". */
+	// +optional
+	StorageType *string `json:"storageType,omitempty"`
+
+	/* The zone to create the Cloud Bigtable cluster in. Each cluster must have a different zone in the same region. Zones that support Bigtable instances are noted on the Cloud Bigtable locations page. */
+	Zone string `json:"zone"`
+}
+
+// BigtableInstanceStatus defines the config connector machine state of BigtableInstance
+type BigtableInstanceStatus struct {
+	/* Conditions represent the latest available observations of the
+	   object's current state. */
+	Conditions []v1alpha1.Condition `json:"conditions,omitempty"`
+
+	/* ObservedGeneration is the generation of the resource that was most recently observed by the Config Connector controller. If this is equal to metadata.generation, then that means that the current reported status reflects the most recent desired state of the resource. */
+	// +optional
+	ObservedGeneration *int64 `json:"observedGeneration,omitempty"`
+
+	/* NOTYET
+	// A unique specifier for the BigtableInstance resource in GCP.
+	// +optional
+	ExternalRef *string `json:"externalRef,omitempty"`
+	*/
+
+	/* NOTYET
+	// ObservedState is the state of the resource as most recently observed in GCP.
+	// +optional
+	ObservedState *BigtableInstanceObservedState `json:"observedState,omitempty"`
+	*/
+}
+
+/* NOTYET
```

Comment:
If not yet, when?

Reply:
So typically we / I have started doing one phase where we get the CRD yaml to match the existing CRD yaml.  It's not a perfect match, the descriptions aren't exact, and we now mark status.observedGeneration as an int64.  But there should be no field changes at this stage (we aren't using a new controller yet)

NOTYET is how I "stash" the fields that are going to be useful as we update the CRD, but we aren't yet adding support.

Reply:
For this instance, we aren't introducing observedState yet, because it's an existing v1beta1 CRD.  The only change we know we need right now for BigtableInstance is around the problems on autoscaling vs numNodes.

Reply:
Not against having a TODO/NOTYET but I do think we should communicate when we would want to include it.
If we want other people to contribute, having sign posts which indicate if this fields is ready to be exposed (or not) will help.
It can indicate its time or that not to waste your effort yet.

Reply:
Any suggestions?  We at least want to wait till we have the direct controller.  Maybe `TODO-AFTER-DIRECT-CONTROLLER` ?

Reply:
Sure. I'd even be happy with something that said something like we want the direct controller to have a minimum 3 month burn in as a beta resource before we add fields the non direct controller did not support.